python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,453 @@
+from pyiceberg.table.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
+from pyiceberg.table.sorting import UNSORTED_SORT_ORDER, SortOrder
+from pyiceberg.typedef import EMPTY_DICT
+
+EXTERNAL_TABLE_TYPE = "EXTERNAL_TABLE"
+
+PROP_GLUE_TABLE = "Table"
+PROP_GLUE_TABLE_TYPE = "TableType"
+PROP_GLUE_TABLE_DESCRIPTION = "description"
+PROP_GLUE_TABLE_PARAMETERS = "Parameters"
+PROP_GLUE_TABLE_DATABASE_NAME = "DatabaseName"
+PROP_GLUE_TABLE_NAME = "Name"
+
+PROP_GLUE_TABLELIST = "TableList"
+
+PROP_GLUE_DATABASE = "Database"
+PROP_GLUE_DATABASE_LIST = "DatabaseList"
+PROP_GLUE_DATABASE_NAME = "Name"
+PROP_GLUE_DATABASE_LOCATION = "LocationUri"
+PROP_GLUE_DATABASE_DESCRIPTION = "Description"
+PROP_GLUE_DATABASE_PARAMETERS = "Parameters"
+
+PROP_GLUE_NEXT_TOKEN = "NextToken"
+
+GLUE_DESCRIPTION_KEY = "comment"
+GLUE_DATABASE_LOCATION_KEY = "location"
+
+
+def _construct_parameters(metadata_location: str) -> Properties:
+    return {TABLE_TYPE: ICEBERG.upper(), METADATA_LOCATION: metadata_location}
+
+
+def _construct_table_input(table_name: str, metadata_location: str, properties: Properties) -> Dict[str, Any]:
+    table_input = {
+        PROP_GLUE_TABLE_NAME: table_name,
+        PROP_GLUE_TABLE_TYPE: EXTERNAL_TABLE_TYPE,
+        PROP_GLUE_TABLE_PARAMETERS: _construct_parameters(metadata_location),
+    }
+
+    if table_description := properties.get(GLUE_DESCRIPTION_KEY):
+        table_input[PROP_GLUE_TABLE_DESCRIPTION] = table_description
+
+    return table_input
+
+
+def _construct_database_input(database_name: str, properties: Properties) -> Dict[str, Any]:
+    database_input: Dict[str, Any] = {PROP_GLUE_DATABASE_NAME: database_name}
+    parameters = {}
+    for k, v in properties.items():
+        if k == GLUE_DESCRIPTION_KEY:
+            database_input[PROP_GLUE_DATABASE_DESCRIPTION] = v
+        if k == GLUE_DATABASE_LOCATION_KEY:
+            database_input[PROP_GLUE_DATABASE_LOCATION] = v
+        else:
+            parameters[k] = v
+    database_input[PROP_GLUE_DATABASE_PARAMETERS] = parameters
+    return database_input
+
+
+def _write_metadata(metadata: TableMetadata, io: FileIO, metadate_path: str):
+    ToOutputFile.table_metadata(metadata, io.new_output(metadate_path))
+
+
+class GlueCatalog(Catalog):
+    def __init__(self, name: str, **properties: str):
+        super().__init__(name, **properties)
+        self.glue = boto3.client("glue")
+
+    def _convert_glue_to_iceberg(self, glue_table: Dict[str, Any]) -> Table:
+        properties: Properties = glue_table[PROP_GLUE_TABLE_PARAMETERS]
+
+        if TABLE_TYPE not in properties:
+            raise NoSuchPropertyException(
+                f"Property {TABLE_TYPE} missing, could not determine type: "
+                f"{glue_table[PROP_GLUE_TABLE_DATABASE_NAME]}.{glue_table[PROP_GLUE_TABLE_NAME]}"
+            )
+        glue_table_type = properties[TABLE_TYPE]
+
+        if glue_table_type.lower() != ICEBERG:
```

Comment:
Thank you for your suggestion. Could you please expand this case further? I guess for "empty" here you mean `glue_table_type == ""` since we raise an exception if `TABLE_TYPE` is not in `properties`.  I wonder if there is some special exception other than `NoSuchIcebergTableError` should be raised when `glue_table_type == ""`